lgc/patch/Patch.cpp

```diff
@@ -241,6 +241,15 @@ void Patch::addPasses(PipelineState *pipelineState, lgc::PassManager &passMgr, b
   }
 }
 
+// =====================================================================================================================
+// Register all the patching passes into the given pass manager
+//
+// @param [in/out] passMgr : Pass manager
+void Patch::registerPasses(lgc::PassManager &passMgr) {
+#define LLPC_PASS(NAME, CLASS) passMgr.registerPass(NAME, decltype(CLASS)::name());
```

Comment:
You are undefing this in the .def file, so it probably makes sense to move the define also to the .def file.

Reply:
@tsymalla, like @Flakebi  says, defining the macro outside of the .def file allows to easily generate different code if we need to in the future, and this is the pattern used in LLVM (including for the pass registry files). I think the reason the undef is in the .def file is that regardless of the code generated when including the .def file, this exact undef will be needed, so by putting it in the .def file, there is no need to write it each time the file is included.

Reply:
I see. I was not aware of that being a pattern in LLVM. Thanks for clarifying.